### PR TITLE
Exclude appeals that require(d) IHPs for the purposes of calculating …

### PR DESCRIPTION
**Pull request by cmgiven**
Noticed during testing that `DirectReviewDocket.new.time_until_due_of_oldest_appeal` was returning 0. Turns out that the receipt dates of the oldest ten appeals are: [Thu, 05 Apr 2018, Tue, 10 Apr 2018, Wed, 16 May 2018, Fri, 01 Jun 2018, Thu, 07 Jun 2018, Sat, 30 Jun 2018, Sun, 01 Jul 2018, Tue, 31 Jul 2018, Sun, 05 Aug 2018, Mon, 06 Aug 2018]

This is because they're blocked by IHPs, which the Board has yet to negotiate deadlines for.

The obvious fix is just to exclude non-ready appeals. This PR does that.

But when an IHP is completed for one of these old appeals, it will become ready, and the next time someone requests a distribution, the docket will suddenly appear dramatically older than it in fact is, overdistributing direct review appeals.

So this PR further proposes excluding appeals that have ever had an IHP task from the calculation.

I'll be talking this through with LRP in a bit. But wanted to get the PR in early.

### CODE REVIEW

File: app/models/appeal.rb [hunk 1]
```diff
@@ -52,6 +52,13 @@ class Appeal < DecisionReview
               DistributionTask.name, Constants.TASK_STATUSES.assigned, 1)
   }
 
+  scope :non_ihp, lambda {
+    joins(:tasks)
+      .group("appeals.id")
+      .having("count(case when tasks.type = ? then 1 end) >= ?",
+              InformalHearingPresentationTask.name, 1)
```
Comment (amprokop) [inline app/models/appeal.rb]: Should this line read `InformalHearingPresentationTask.name, 0)`

**Comment by amprokop** [comment]
Sounds good on the approach — once we enforce an expiration time on IHPs, will this still be a problem?

File: app/models/appeal.rb (hunk 1 above)
Comment (cmgiven) [reply app/models/appeal.rb]: It's greater than or equal to, so 1 is what we're looking for. At least 1 task.

**Comment by cmgiven** [comment]
That is the open question. If there is an expiration time, then yes, this is not a problem, although this fix probably doesn't cause harm in that world either.

File: app/models/appeal.rb (hunk 1 above)
Comment (cmgiven) [reply app/models/appeal.rb]: Oh, shoot, it should be equal to 0. D'oh.

**Pull request merged**